tribestream-api-registry-webapp/src/main/java/org/tomitribe/tribestream/registryng/resources/ApplicationResource.java

```diff
@@ -89,11 +90,17 @@ public Response createService(
             ApplicationWrapper application) {
         final Swagger swagger = application.getSwagger();
         validate(swagger);
-        final OpenApiDocument document = repository.insert(swagger);
-        final OpenApiDocument newDocument = repository.findByApplicationIdWithEndpoints(document.getId());
-        return Response.status(Response.Status.CREATED)
-                .entity(processor.toWrapper(newDocument, linker.buildApplicationLinks(uriInfo, newDocument)))
-                .build();
+        try {
+            final OpenApiDocument document = repository.insert(swagger);
+            final OpenApiDocument newDocument = repository.findByApplicationIdWithEndpoints(document.getId());
+            return Response.status(Response.Status.CREATED)
+                    .entity(processor.toWrapper(newDocument, linker.buildApplicationLinks(uriInfo, newDocument)))
+                    .build();
+        } catch (TransactionalException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
Good question. I was thinking about changing the service itself, but didn't manage to get the transitional annotation to catch the exception from the em.persist method. hibernate tries to commit the change after the service method is done, so no way to catch an exception here...

https://github.com/tomitribe/tribestream-api-registry/blob/master/tribestream-api-registry-webapp/src/main/java/org/tomitribe/tribestream/registryng/repository/Repository.java#L275

please advise.